### PR TITLE
Install script improvements

### PR DESCRIPTION
**Pull request by tfalders**
Ports some of the nice features of rocSOLVER's install script to hipSOLVER, including the cmake-arg argument and the ability to specify relative paths for custom rocBLAS and rocSOLVER locations.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -304,6 +310,8 @@ build_static=false
 build_release_debug=false
 build_codecoverage=false
 build_address_sanitizer=false
+declare -a cmake_common_options
+declare -a cmake_client_options
```
Comment (cgmb) [inline install.sh:314]: You're declaring these variables as arrays (which is good) and using array append for the --cmake-arg flag (`+=("element"`), but still using string concatenation elsewhere (e.g., `cmake_common_options="${cmake_common_options} element"`). The first string concatenation is going to change the variable type from array to string.

**Comment by cgmb** [comment]
The CI on hipSOLVER is cheap and it does at least run the install script.

File: install.sh (hunk 1 above)
Comment (tfalders) [reply install.sh:314]: Ah, my ignorance of shell scripting is revealed! I'll update the rest of the cmake_common_options additions.

File: install.sh [hunk 2]
```diff
@@ -462,55 +481,53 @@ pushd .
...
+    cmake_common_options+=("-DUSE_CUDA=ON")
   else
-    cmake_common_options="${cmake_common_options} -DUSE_CUDA=OFF"
+    cmake_common_options+=("-DUSE_CUDA=OFF")
   fi
 
   # clients
   if [[ "${build_clients}" == true ]]; then
-    cmake_client_options="${cmake_client_options} -DBUILD_CLIENTS_SAMPLES=ON -DBUILD_CLIENTS_TESTS=ON -DBUILD_CLIENTS_BENCHMARKS=ON"
+    cmake_client_options+=("-DBUILD_CLIENTS_SAMPLES=ON -DBUILD_CLIENTS_TESTS=ON -DBUILD_CLIENTS_BENCHMARKS=ON")
```
Comment (cgmb) [inline install.sh:516]: Almost. Take a look at how it was done in rocSOLVER. What you've done works, but the value of using an array is that you can correctly handle arguments that contain spaces. (Not that we do, but it becomes possible.)

When you add multiple arguments separated by spaces in a single array element, you then need to split on spaces later, which breaks the original reason for using the array.

Comment (tfalders) [reply install.sh:516]: I tried to split up this string like in rocSOLVER, but it didn't seem to work properly for me. I can give it another try today.

**Review by jzuniga-amd — APPROVED** [review]
LGTM

File: install.sh (hunk 2 above)
Comment (tfalders) [reply install.sh]: Looks like I missed the `[@]` where cmake_common_options and cmake_client_options are used. Separating the array elements with spaces seems to work now.

**Review by cgmb — APPROVED** [review]

**Pull request merged**